Show AudioEffectCapture by use of a shader showing the captured waveform in stereo

goatchurchprime opened the pull request:
Microphone data is normally read using AudioEffectCapture rather than AudioEffectRecord, so we should adapt this demo project to use it.  And a good way to show that something is happening is to plot the waveform using a shader.  

This shows it in stereo because a lot of computers have two microphones.  

It's also useful to measure the sample rate of the microphone according to the data stream it is generating.

![Screenshot from 2025-03-02 18-23-26](https://github.com/user-attachments/assets/b3162b01-0bd6-4277-9446-1f238eadbcdb)

Calinou commented on audio/mic_input/MicRecord.gd:
```diff
@@ -0,0 +1,123 @@
+var mix_rate := 44100  # This is the default mix rate on recordings.
+var format := AudioStreamWAV.FORMAT_16_BITS  # This is the default format on recordings.
+
+var audiosamplesize : int = 882
+var audiosampleframetextureimage : Image
+var audiosampleframetexture : ImageTexture
+var totalsamples = 0
+var sampleduration = 0.0
+var recordingbuffer = null
```
Use `snake_case` for variable names, e.g. `audio_sample_size`.

Calinou commented on audio/mic_input/MicRecord.gd:
```diff
@@ -0,0 +1,123 @@
+	totalsamples = 0
+	sampleduration = 0.0
+	if recordingbuffer != null:
+		#recording = recordeffect.get_recording()
+		$PlayButton.disabled = false
+		$SaveButton.disabled = false
+		#recordeffect.set_recording_active(false)
+		#recording.set_mix_rate(mix_rate)
+		#recording.set_format(format)
+		#recording.set_stereo(stereo)
```
Remove commented out lines.

Calinou reviewed (commented):
Tested locally on 4.4.stable (Linux), it works as expected. I've tested `mic_record`, not `mic_input` which fails with a script error on startup:

```
Parser Error: Static function "start_microphone()" not found in base "GDScriptNativeClass".
```

Is there a reason we have two demo projects now? They basically look identical at a surface, and we should try to keep the existing name for compatibility.

One issue I noticed in `mic_record` (but also `mic_input`) that the root node in the project (`MicRecord`) is rotated by 0.1 degrees, so the whole UI looks very slightly rotated. This is noticeable on the visualizer because it's not 100% straight.

goatchurchprime commented:
I've rolled back the changes to the `audio/mic_record` demo in the hope that this way of accessing the microphone will eventually become obsolete if [PR#105244](https://github.com/godotengine/godot/pull/105244) gets accepted.

I've also added in some shader code to visualize the sum of the left and right stereo microphones when progressively shifted in time.  This is best tested with a pure tone of 400Hz, making a consistent 8 cycles across the 20ms sample window.  To make this easy I've added in a drop-down option to generate these tones.

If you deploy this project to an Android phone so you can play the tone out loud while moving it to different positions about your stereo microphones, you get this kind of movie where you can see the waveforms shifting in amplitude and phase.

[Screencast From 2025-04-16 12-18-26.webm](https://github.com/user-attachments/assets/77f48fc2-2863-4003-ba08-98c0264cc7fd)
